strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-pypi-layout-provider/src/test/java/org/carlspring/strongbox/artifact/generator/PypiArtifactGenerator.java

```diff
@@ -91,43 +96,48 @@ public Path generateArtifact(String id,
                                                                           "none",
                                                                           "any",
                                                                           "whl");
-        return generateArtifact(coordinates);
+        return generateArtifact(coordinates, byteSize);
     }
 
     @Override
     public Path generateArtifact(URI uri,
-                                 long size)
+                                 long byteSize)
             throws IOException
     {
         PypiArtifactCoordinates coordinates = PypiArtifactCoordinates.parse(FilenameUtils.getName(uri.toString()));
-        return generateArtifact(coordinates);
+        return generateArtifact(coordinates, byteSize);
     }
 
-    public Path generateArtifact(PypiArtifactCoordinates coordinates)
+    public Path generateArtifact(PypiArtifactCoordinates coordinates,
+                                 long byteSize)
             throws IOException
     {
         String packagePath = coordinates.toPath();
 
         Path fullPath = basedir.resolve(packagePath);
 
-        try(ZipOutputStream zos = new ZipOutputStream(Files.newOutputStream(fullPath)))
+        new File(fullPath.getParent().toString()).mkdirs();
```

Comment:
Please, use Java NIO.2 when possible:
```suggestion
        Path fullPathParent = fullPath.getParent();
        Files.createDirectories(fullPathParent);
```